src/EFCore.Design/Scaffolding/Internal/RelationalScaffoldingModelFactory.cs

```diff
@@ -166,7 +166,10 @@ protected virtual ModelBuilder VisitDatabaseModel(ModelBuilder modelBuilder, Dat
 
         if (!string.IsNullOrEmpty(databaseModel.DatabaseName))
         {
-            modelBuilder.Model.SetDatabaseName(databaseModel.DatabaseName);
+            modelBuilder.Model.SetDatabaseName(
+                            _options.UseDatabaseNames && !string.IsNullOrEmpty(databaseModel.DatabaseName)
```

Comment:
```suggestion
                            !_options.UseDatabaseNames && !string.IsNullOrEmpty(databaseModel.DatabaseName)
```

`_options.UseDatabaseNames` is a bit confusing in this context, but it means to use the names of things from the database directly without changing their case or convention. In other words, your tests are backward. `true` should not change the originals while `false` should capitalize and make them Pascal case.

Reply:
Okay, thank you very much for pointing it out!

Reply:
lol, it looks like I mislead you in two previous comments. 🙂 So really, you implemented it in perfect compliance to my faulty description.